Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1461,6 +1461,44 @@ std::optional<LayoutUnit> RenderGrid::firstLineBaseline() const
     return baseline.value() + baselineChild->logicalTop().toInt();
 }
 
+std::optional<LayoutUnit> RenderGrid::lastLineBaseline() const
+{
+    if (isWritingModeRoot() || !m_grid.hasGridItems() || shouldApplyLayoutContainment())
+        return std::nullopt;
+
+    const RenderBox* baselineChild = nullptr;
+    unsigned numColumns = m_grid.numTracks(ForColumns);
+    unsigned numRows = m_grid.numTracks(ForRows);
+
+    for (int column = numColumns - 1; column > -1; column--) {
```

Comment:
Should we always be iterating through rows, until we find one that has items?

It looks like `firstLineBaseline` also doesn't do this though, so probably no need to fix this right now.

These two functions are really similar, is it possible to share code for them? The original has some 'FIXME' comments that would be nice to have applied to both.

Reply:
I added RenderGrid::getBaselineChild to help out with this. Let me know what you think! We could also try to modify synthesizedBaselineFromBorderBox so that it gets the direction on its own instead of passing it in if we want to try to clean it up further.